### ISSUE TITLE
prefer_final_in_for_each false positives

### ISSUE THREAD

**Issue by davidmorgan** [dart-lang/linter#2564]
Where there are multiple declarations and one or more is reassigned, the lint should allow var

```
for (var x = 0, y = 0; x != 10; ++x) {
  ++y;
}
```

**Comment by oprypin** [comment]
It seems that this bug is obsolete.

Current state is:

```dart
  for (var x = 0, y = 10; x < y; ++x) { // OK
    print(y);
  }
```
```dart
  var x = 0;
  for (var y = 10; x < y; ++x) {  // prefer_final_locals
    print(y);
  }
```
